File: Intensive/CPU/cpu.cpp
#include <iostream>
#include <chrono>
#include <cmath>

void cpu_intensive_test() {
    const long long iterations = 1000000000; // 1B
    double sum = 0.0;
    const double pi = 3.14159265358979323846;
    auto start = std::chrono::high_resolution_clock::now();
    for (long long i = 0; i < iterations; ++i) {
        double x = (i * pi) / 180.0;
        sum += std::sin(x) * std::cos(x);
    }
    auto end = std::chrono::high_resolution_clock::now();
    std::cout << "CPU-intensive test: sum = " << sum 
              << ", time elapsed " << std::chrono::duration<double>(end - start).count() 
              << " seconds.\n";
}

int main() {
    for (size_t i = 0; i < 1; ++i) {
        cpu_intensive_test();
    }
    return 0;
}

File: Intensive/CPU/cpu.rs
use std::time::Instant;
use std::f64::consts::PI;

fn cpu_intensive_test() {
    let iterations = 1_000_000_000; // 1B
    let mut sum = 0.0;
    let start = Instant::now();
    for i in 0..iterations {
        let x = (i as f64) * PI / 180.0;
        sum += x.sin() * x.cos();
    }
    println!("CPU-intensive test: sum = {}, time elapsed {:?}", sum, start.elapsed());
}

fn main() {
    for _ in 0..1 {
        cpu_intensive_test();
    }
}
